File: MyLanguageWeb/src/main/resources/static/js/classes/dto/entity/customer_collection/response/customer_collection_response_dto.js
import {
    CustomerResponseDTO
} from "../../customer/response/customer_response_dto.js";

import {
    LangResponseDTO
} from "../../lang/response/lang_response_dto.js";

export class CustomerCollectionResponseDTO {
    #id;
    #title;
    #numberOfWords;
    #isActiveForAuthor;
    #pathToImage;
    #description;
    #dateOfCreate;
    #customer;
    #lang;

    constructor(json) {
        if (json) {
            this.#id = json["id"];
            this.#title = json["title"];
            this.#numberOfWords = json["number_of_words"];
            this.#isActiveForAuthor = json["is_active_for_author"];
            this.#pathToImage = json["path_to_image"];
            this.#description = json["description"];

            let dateOfCreate = json["date_of_create"];
            if (dateOfCreate) {
                this.#dateOfCreate = new Date(dateOfCreate);
            }

            let customer = json["customer"];
            if (customer) {
                this.#customer = new CustomerResponseDTO(customer);
            }

            let lang = json["lang"];
            if (lang) {
                this.#lang = new LangResponseDTO(lang);
            }
        }
    }

    getId() {
        return this.#id;
    }

    getTitle() {
        return this.#title;
    }

    getDateOfCreate() {
        return this.#dateOfCreate;
    }

    getNumberOfWords() {
        return this.#numberOfWords;
    }

    getIsActiveForAuthor() {
        return this.#isActiveForAuthor;
    }

    getPathToImage() {
        return this.#pathToImage;
    }

    getDescription() {
        return this.#description;
    }

    getCustomer() {
        return this.#customer;
    }

    getLang() {
        return this.#lang;
    }
}

File: MyLanguageWeb/src/main/java/ru/herooo/mylanguageweb/dto/entity/customercollection/response/CustomerCollectionResponseDTO.java
package ru.herooo.mylanguageweb.dto.entity.customercollection.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import ru.herooo.mylanguageweb.controllers.json.LongDeserializer;
import ru.herooo.mylanguageweb.controllers.json.LongSerializer;
import ru.herooo.mylanguageweb.dto.entity.customer.response.CustomerResponseDTO;
import ru.herooo.mylanguageweb.dto.entity.lang.response.LangResponseDTO;

import java.time.LocalDateTime;

public class CustomerCollectionResponseDTO {
    @JsonSerialize(using = LongSerializer.class)
    @JsonDeserialize(using = LongDeserializer.class)
    @JsonProperty("id")
    private long id;

    @JsonProperty("title")
    private String title;

    @JsonProperty("date_of_create")
    private LocalDateTime dateOfCreate;

    @JsonProperty("number_of_words")
    private long numberOfWords;

    @JsonProperty("is_active_for_author")
    private boolean isActiveForAuthor;

    @JsonProperty("customer")
    private CustomerResponseDTO customer;

    @JsonProperty("lang")
    private LangResponseDTO lang;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public LocalDateTime getDateOfCreate() {
        return dateOfCreate;
    }

    public void setDateOfCreate(LocalDateTime dateOfCreate) {
        this.dateOfCreate = dateOfCreate;
    }

    public CustomerResponseDTO getCustomer() {
        return customer;
    }

    public void setCustomer(CustomerResponseDTO customer) {
        this.customer = customer;
    }

    public LangResponseDTO getLang() {
        return lang;
    }

    public void setLang(LangResponseDTO lang) {
        this.lang = lang;
    }

    public boolean getIsActiveForAuthor() {
        return isActiveForAuthor;
    }

    public void setIsActiveForAuthor(boolean activeForAuthor) {
        isActiveForAuthor = activeForAuthor;
    }

    public long getNumberOfWords() {
        return numberOfWords;
    }

    public void setNumberOfWords(long numberOfWords) {
        this.numberOfWords = numberOfWords;
    }
}
